### PR TITLE
statistics: no longer pass inAndEqCnt to calc pseudo index row count

### PR DESCRIPTION
**Pull request by winoros**
After support calculating range for DNF expression of index in the near future, inAndEqCnt is not easy to maintain, so i decide calulate here rather than pass it in method.
@shenli @hanfei1991 PTAL

### CODE REVIEW

File: statistics/table.go [hunk 1]
```diff
@@ -237,17 +237,29 @@ func PseudoTable(tableID int64) *Table {
...
-		i := len(indexRange.LowVal) - 1
-		if i > inAndEQCnt {
-			i = inAndEQCnt
+		var i int
+		if len(indexRange.LowVal) < len(indexRange.HighVal) {
+			i = len(indexRange.LowVal) - 1
+		} else {
+			i = len(indexRange.HighVal) - 1
+		}
+		for pos := 0; pos <= i; pos++ {
```
Comment (hanfei1991) [inline statistics/table.go]: Extract it as a function.

File: statistics/table.go [hunk 2]
```diff
@@ -237,17 +237,29 @@ func PseudoTable(tableID int64) *Table {
...
 		return 0, nil
 	}
 	var totalCount float64
 	for _, indexRange := range indexRanges {
 		count := tableRowCount
-		i := len(indexRange.LowVal) - 1
-		if i > inAndEQCnt {
-			i = inAndEQCnt
+		var i int
+		if len(indexRange.LowVal) < len(indexRange.HighVal) {
```
Comment (hanfei1991) [inline statistics/table.go]: I think these two len are always the same.

**Comment by hanfei1991** [comment]
LGTM

**Comment by hanfei1991** [comment]
@shenli PTAL

File: util/types/range.go [hunk 3]
```diff
@@ -142,6 +143,22 @@ func (ir *IndexRange) Align(numColumns int) {
 	}
 }
 
+// PrefixEqualLen tells you how long the prefix of the range is a point.
+// e.g. If this range is (1 2 3, 1 2 +inf), then the return value is 2.
+func (ir *IndexRange) PrefixEqualLen(sc *variable.StatementContext) (int, error) {
+	// Here, len(ir.LowVal) always equal to len(ir.HighVal)
+	for i := 0; i < len(ir.LowVal); i++ {
+		cmp, err := ir.LowVal[i].CompareDatum(sc, ir.HighVal[i])
```
Comment (shenli) [inline util/types/range.go]: Is it possible that len(ir.LowVal) > len(ir.HighVal)?

Comment (shenli) [inline util/types/range.go]: Got it! I just notice the comment.

**Comment by shenli** [comment]
LGTM

**Pull request merged**